### ISSUE TITLE
KernelTestCase consumes too much time for cache cheking

### ISSUE THREAD

**Issue by tai2** [symfony/symfony#16989]
In our project, automated tests are growing day by day and those are slowing our development speed down. 

So I took a profile for the project. I found *ConfigCache$isFresh()* consumes 60% of kernel booting for each test method. This includes huge number of function calls for *file_exists()* and *filemtime()*. 

Then, I tried setting debug option false when kernel booting. The result was good, improving whole test speed. But it is a bad idea that disabling debug mode during development time. This doesn't solve the problem.

So could you consider my idea that checking the cache only once for the first test case invocation? Container setting should not change during tests are running, so check is not necessary for every time. This is the internal of symfony kernel, so I don't know how do I bypass this problem.

version: symfony 2.7

**Comment by stof** [comment]
Well, the debug mode of the kernel precisely controls whether the cache is checked for freshness of no. This is almost the *only* thing which depends directly on the debug mode (many config settings have a default value which depends on this flag, but it is only a default configuration).

So using the kernel in no-debug mode in tests *is* the way to make your testsuite faster in this case.

**Comment by tai2** [comment]
Thank you for a reply.

>This is almost the only thing which depends directly on the debug mode 
It sounds good for me.

Our ordinal workflow is here:

1. Implement some features or fix some bugs.
2. Run tests.
3. Go to 1.

Now let's assume we run the tests with no-debug mode.
When we add some *services* to the container at step 1, the tests will fail because cached settings are used. This can make developers confused.

So if we use no-debug mode, we should clear the caches before step 2, and it's better working automatically. Do you have any ideas?

**Comment by stof** [comment]
@tai2 you can totally have something in your testsuite clearing the cache at the beginning of the whole suite

**Comment by tai2** [comment]
Okay, I'll try it. Thanks!

**Comment by jakzal** [comment]
@tai2 can we close this issue?

**Comment by tai2** [comment]
@jakzal Yes. I'll make another issue if I meet a problem.
